sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala

```diff
@@ -845,6 +859,96 @@ class RocksDBSuite extends AlsoTestWithChangelogCheckpointingEnabled with Shared
     }
   }
 
+  testWithChangelogCheckpointingEnabled("ensure merge operation is not supported" +
+    " with changelog checkpoint if column families is not enabled") {
+    withTempDir { dir =>
+      val remoteDir = Utils.createTempDir().toString
+      val conf = dbConf.copy(minDeltasForSnapshot = 5, compactOnCommit = false)
+      new File(remoteDir).delete() // to make sure that the directory gets created
+      withDB(remoteDir, conf = conf, useColumnFamilies = false) { db =>
+        db.load(0)
+        db.put("a", "1")
+        intercept[UnsupportedOperationException](
+          db.merge("a", "2")
+        )
+      }
+    }
+  }
+
+  testWithChangelogCheckpointingDisabled("ensure merge operation is supported" +
+    " without changelog checkpoint if column families is not enabled") {
```

Comment:
Hmm - why do we have these tests ? Which case are we trying to cover here ?

Reply:
This testcase is trying to cover the scenario where merge is used without enabling column families. I removed the changelog checkpoint condition, as we made enabling column families a pre-requisite for merge.

Reply:
So should we update the test to run with changelog checkpointing enabled and disabled both ?

Reply:
yeah, I did update the test to run with both changelog checkpoint enabled and disabled. Actually, we don't need this second test now. I have removed it. The test above covers it.